velero-plugin-for-aws/object_store.go

```diff
@@ -145,8 +146,23 @@ func (o *ObjectStore) Init(config map[string]string) error {
 
 	if insecureSkipTLSVerify {
 		serverConfig.HTTPClient = &http.Client{
+			// Copied from net/http
```

Comment:
I guess the other option here is rather then constructing a new `*http.Client` and `*http.Transport`, you could type-assert the existing `serverConfig.HTTPClient.Transport` to a `*http.Transport`, and then you can just set the `TLSClientConfig` field. Might be a little more maintainable if any defaults change.

I think it's fine either way, so whichever approach you prefer works for me.

Reply:
It looks like this PR used this approach already: https://github.com/vmware-tanzu/velero/pull/2364/files#diff-228685d15e46dc75136c60078abb0e35R118

Reply:
@nrb pointed out that `serverConfig.HTTPClient.Transport` may not actually use the default transport, so I guess you would instantiate the default transport, type-assert it, edit the TLSClientConfig, and then use it in the http.Client.

Reply:
Thank you I will update this. I actually only did what I did because it's the same code used in Restic. Wasn't a huge fan of just copying it out of the net lib.